django/core/management/__init__.py

```diff
@@ -378,6 +385,24 @@ def execute(self):
         except CommandError:
             pass  # Ignore any option errors at this point.
 
+        # Read the settings module from the local pyproject.toml if present.
+        pyproject = Path("pyproject.toml")
+        if pyproject.is_file():
+            with pyproject.open("rb") as f:
+                _config = tomllib.load(f)
+            _settings_module = (
+                _config.get("tool", {}).get("django", {}).get("settings_module")
+            )
+            if _settings_module is not None:
+                os.environ.setdefault("DJANGO_SETTINGS_MODULE", _settings_module)
+                if _settings_module == os.environ["DJANGO_SETTINGS_MODULE"]:
+                    # The runserver command starts up a subprocess that has
+                    # DJANGO_SETTINGS_MODULE set, so we can't only add to
+                    # the path if DJANGO_SETTINGS_MODULE is not already set.
+                    #
+                    # Insert the pyproject.toml directory into the path.
+                    sys.path.insert(0, os.getcwd())
```

Comment:
We might be able to punt on it for the first iteration, but we want the path to be configurable from the pyproject.toml as well.